sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -661,6 +661,36 @@ class Analyzer(
       // rule: ResolveDeserializer.
       case plan if containsDeserializer(plan.expressions) => plan
 
+      // Some commands (AlterTableDropPartitionCommand) use expressions, so we need to resolve.
+      case c: CommandWithExpression =>
+        val catalogTable = catalog.getTableMetadata(c.getTableName)
+
+        // We should not do this here in order to support general CommandWithExpression
+        if (catalogTable.tableType == CatalogTableType.VIEW) {
+          throw new AnalysisException(
+            "Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead")
```

Comment:
Yes. There is no such command.
`AlterTableDropPartitionCommand` is not visible in `catalyst`. So, it's inevitable to create some corresponding one in `catalyst`.

This is an real issue. So, @rxin told me it's a weird. 
Until now, I didn't find a better way to support expressions on `Command`.
I'm asking the committers not to use expressions here.